encoding_negotiator.js

```diff
@@ -0,0 +1,31 @@
+var zlib = require('zlib')
+var Negotiator = require('negotiator')
+
+/**
+ * @const
+ * whether current node version has brotli support
+ */
+var hasBrotliSupport = 'createBrotliCompress' in zlib
+
+function negotiateEncoding (req, encodings_) {
```

Comment:
This is a good change and I'd skip `accepts` update, I don't think I'm prepared to maintain multiple release lines as I mentioned in your PR to `accepts` (one for old node for this package, then a new major version with improved API).

I'd just use this directly, and simplify the code to:

```js
function negotiateEncoding (req, encodings) {
  var negotiator = new Negotiator(req)
  return negotiator.encodings(encodings, hasBrotliSupport ? ['br'] : ['gzip'])[0] || false
}
```

Reply:
This wouldn't introduce a breaking change by not supporting multiple strings, would it? Maybe we should put a deprecation message

Reply:
Why would you be deprecating an internal API that isn't exposed?

Reply:
The is the only place the code is used: https://github.com/expressjs/compression/blob/5e6ee06a510170088615b11a61cef9faefca06e1/index.js#L177-L183

Nothing here changes for a user.

Reply:
I'm not too familiar (yet) with both libraries, but I suspect that code can also be simplified since the `deflate` issue sounds related to preferred?

Reply:
You are right, there is no breaking change

Reply:
I'd be happy to open a separate PR removing `accepts` if you're interested? Or are you ok landing as part of this PR?

Reply:
I think it would be better to remove `accepts` now that we're using `negotiation`.

Reply:
OK, in which case the code could be further simplified to use the preferred option for both `deflate` and `brotli`, and `gzip`. And I think you might want to initialize `new Negotiator` once, instead of each time in that method.

Reply:
> I'm not too familiar (yet) with both libraries, but I suspect that code can also be simplified since the deflate issue sounds related to preferred?

I don't remember the exact reason (while I was researching more about the decisions that had been made), but I think it wasn't due to preference but for another reason. I believe it was because deflate was slower, so the former maintainer had decided that gzip was always preferred over deflate

Reply:
Sorry, I meant the merged `preferred` option in negotiator, i.e. can trim the extra lines now and pass `['gzip']` as the preferred option to avoid the secondary check. I think it's why a separate PR here is a bit simpler to review: https://github.com/expressjs/compression/pull/197.